src/components/JoinPage/ EmailAuthentication/index.tsx

```diff
@@ -1,18 +1,53 @@
+import { useRef, useState } from "react";
+
+import ErrorPopup from "../ErrorPopup";
 import { St } from "./style";
 
 export default function EmailAuthentication() {
+  const [emailText, setEmailText] = useState<string>("");
+  const [isPopupOpen, setIsPopupOpen] = useState<any>(false);
+  const emailInput = useRef<HTMLInputElement | null>(null);
+
+  const getUserEmail = () => {
+    console.log(emailInput.current);
+    emailInput.current && setEmailText(emailInput.current.value);
+    console.log(emailText);
+  };
+
+  const verifyEmail = () => {
+    const email: string | null = emailInput.current && emailInput.current.value;
```

Comment:
null 타입을 옵셔널로 정의하는 이유는 무엇인가요?

Reply:
email이 null값이라고 뜰 때도 있어서 그랬던 것 같아요! 그래서 다음 코드에 if문에도 email값이 없을때를 가정해줬었는데, 아마 onchange로 email을 받아와서 그런 것 같아요! 위에 로직 고치면서 수정할께요😁

Reply:
갑자기 궁금해졌는데, `null 타입을 옵셔널로 정의하는 이유는 무엇인가요?`질문의 이유가 궁금해요!
email에 null값이 오면 안된다고 생각해서인지, 아니면 null값이 올리가 없다고 생각해서인지 아니면 다른이유인지 아니면 그냥인지!

Reply:
👀반영 제안👀

오호! 당시에 null 값을 포함한 변수 선언이 조금 어색했어요!
그래서 아래와 같은 코드를 생각하고 물어봤던 것이었습니다!

```suggestion
    if (!emailInput.current) return;
    
    const email: string =  emailInput.current.value;
```

이렇게 하면 email 이 null 임을 계속 확인할 필요 없으니까요!